### PR TITLE
Feature/core 10304 add geo tree customization

### PR DESCRIPTION
**Pull request by Navas23**
Finally I had to use folder prop instead of slots as we would have to also modify geo-list-item to allow icon as slots

### CODE REVIEW

File: src/elements/GeoTree/GeoTreeItem.vue [hunk 1]
```diff
@@ -1,13 +1,12 @@
 <template>
   <li
     class="geo-tree-item"
-    @mouseover="isShowActionButton = true"
```
Comment (fofi) [inline src/elements/GeoTree/GeoTreeItem.vue:4]: Why do we want to remove this?

**Comment by Navas23** [comment]
The problem of the 2 props is that we could not have those icons in the DS... So I will keep it like this but changing the prop name

File: src/elements/GeoTree/GeoTreeItem.vue [hunk 2]
```diff
@@ -128,6 +138,14 @@ export default {
     expandedCategories: {
       type: Object,
       required: true
+    },
+    /**
+     * Use folder icon instead of chevron
+     */
+    useFolderIcon: {
```
[the remark this answers is not in the document]
Comment (Navas23) [reply src/elements/GeoTree/GeoTreeItem.vue:145]: WDYT about isUsingFolderIcon

File: src/elements/GeoTree/GeoTreeItem.vue (hunk 1 above)
Comment (fofi) [reply src/elements/GeoTree/GeoTreeItem.vue:4]: And why before we had this? are we going to break something?

Comment (Navas23) [reply src/elements/GeoTree/GeoTreeItem.vue:4]: IDK, we were not using it (in fact it was not usable as we had no props in the slots so we could not know which element we clicked)

File: src/elements/GeoTree/GeoTreeItem.vue [hunk 3]
```diff
@@ -128,6 +141,26 @@ export default {
...
+    },
+    /**
+     * Optional Font Awesome 5 icon to use as collapsed icon
+     *
+     * See [vue-fontawesome](https://www.npmjs.com/package/@fortawesome/vue-fontawesome#explicit-prefix-note-the-vue-bind-shorthand-because-this-uses-an-array)
+     * for more info about this.
+     */
+    collapsedIcon: {
+      type: Array,
+      required: false
```
Comment (fofi) [inline src/elements/GeoTree/GeoTreeItem.vue:153]: We could add here the defaultIcon

**Review by fofi — APPROVED** [review]

File: src/elements/GeoTree/GeoTreeItem.vue (hunk 3 above)
Comment (Navas23) [reply src/elements/GeoTree/GeoTreeItem.vue:153]: I didn't add it so I can know if the icon is custom or not (to rotate the icon)

Comment (fofi) [reply src/elements/GeoTree/GeoTreeItem.vue:153]: But, we will rotate the icon if we do not have expandedIcon, right? it has nothing todo with the collapsed one

Comment (Navas23) [reply src/elements/GeoTree/GeoTreeItem.vue:153]: You are right, I will update it

**Review by dletran — APPROVED** [review]

**Review by fofi — APPROVED** [review]

**Pull request merged**